lib/_debug_agent.js

```diff
@@ -81,6 +85,12 @@ Agent.prototype.notifyWait = function notifyWait() {
   this.first = false;
 };
 
+Agent.prototype.destroyAllClients = function destroyAllClients() {
+  this.clients.forEach(function(client) {
+    client.destroy();
```

Comment:
this ends up calling `this.socket.destroy`

I didn't dig too much further, but is this operation synchronous? If not there might be some unexpected behavior here

/cc @bnoordhuis